src/java/main/org/apache/zookeeper/server/quorum/auth/SaslQuorumAuthLearner.java

```diff
@@ -94,7 +94,10 @@ public void authenticate(Socket sock, String hostName) throws IOException {
                     principalConfig,
                     QuorumAuth.QUORUM_SERVER_PROTOCOL_NAME,
                     QuorumAuth.QUORUM_SERVER_SASL_DIGEST, LOG, "QuorumLearner");
-
+            if (sc == null) {
```

Comment:
Same feedback as #495 

1) check the callers and see if it's handled properly. Likely it will be logged there as well. Verify/report.
2) No need to say exception in an exception. The text of LOG.error line seems like it would have been a good error string for the exception itself.
3) as previously noted, add a test.

Reply:
For #1:
Follower#77,Observer#69,QuorumCnxManager#333 all have same patern:
`try { //root caller } catch (IOException e) {//handler code}`
#2 and #3, @brettKK .

Reply:
Hum, it is hard for me to write a unit test for this bug, any suggestion? @brettKK @anmolnar @phunt @nkalmar @others

Reply:
I will try unit test written by @brettKK ~~